system/Common.php

```diff
@@ -270,8 +270,11 @@ function esc($data, $context = 'html', $encoding = null)
 				$method = 'escape' . ucfirst($context);
 			}
 
-			// @todo Optimize this to only load a single instance during page request.
-			$escaper = new \Zend\Escaper\Escaper($encoding);
+                       static $escaper;
```

Comment:
I like the idea of optimizing this, since it will get called quite a few times on a page. However, I think we should check the encoding to make sure they match the static version (using `$escaper->getEncoding()`). If they don't it returns a new instance, otherwise returns the static instance. That way we get the benefit of caching it in 99% of the cases, but still have the flexibility if someone needs to do something strange with it.